module/settings.js

```diff
@@ -52,4 +52,106 @@ export const registerSystemSettings = function () {
+    scope: "world",
+    config: true,
+    type: String,
+    choices: {
+      "no": game.i18n.localize("YZECORIOLIS.SettingRollInfoNo"),
+      "pc": game.i18n.localize("YZECORIOLIS.SettingRollInfoPC"),
+      "npc": game.i18n.localize("YZECORIOLIS.SettingRollInfoNPC"),
+      "all": game.i18n.localize("YZECORIOLIS.SettingRollInfoAll")
+      },
+    default: "no",
+    onChange: applyRollInfoSetting,
+  });
+
+  game.settings.register("yzecoriolis", "RollInfoSkill", {
+    name: game.i18n.localize("YZECORIOLIS.SettingRollInfoSkill"),
+    hint: game.i18n.localize("YZECORIOLIS.SettingRollInfoSkillHint"),
+    scope: "world",
+    config: true,
+    type: String,
+    choices: {
+      "no": game.i18n.localize("YZECORIOLIS.SettingRollInfoNo"),
+      "pc": game.i18n.localize("YZECORIOLIS.SettingRollInfoPC"),
+      "npc": game.i18n.localize("YZECORIOLIS.SettingRollInfoNPC"),
+      "all": game.i18n.localize("YZECORIOLIS.SettingRollInfoAll")
+      },
+    default: "no",
+    onChange: applyRollInfoSetting,
+  });
+
+  game.settings.register("yzecoriolis", "RollInfoWeapon", {
+    name: game.i18n.localize("YZECORIOLIS.SettingRollInfoWeapon"),
+    hint: game.i18n.localize("YZECORIOLIS.SettingRollInfoWeaponHint"),
+    scope: "world",
+    config: true,
+    type: String,
+    choices: {
+      "no": game.i18n.localize("YZECORIOLIS.SettingRollInfoNo"),
+      "pc": game.i18n.localize("YZECORIOLIS.SettingRollInfoPC"),
+      "npc": game.i18n.localize("YZECORIOLIS.SettingRollInfoNPC"),
+      "all": game.i18n.localize("YZECORIOLIS.SettingRollInfoAll")
+      },
+    default: "no",
+    onChange: applyRollInfoSetting,
+  });
+
+  game.settings.register("yzecoriolis", "RollInfoExplosive", {
+    name: game.i18n.localize("YZECORIOLIS.SettingRollInfoExplosive"),
+    hint: game.i18n.localize("YZECORIOLIS.SettingRollInfoExplosiveHint"),
+    scope: "world",
+    config: true,
+    type: String,
+    choices: {
+      "no": game.i18n.localize("YZECORIOLIS.SettingRollInfoNo"),
+      "pc": game.i18n.localize("YZECORIOLIS.SettingRollInfoPC"),
+      "npc": game.i18n.localize("YZECORIOLIS.SettingRollInfoNPC"),
+      "all": game.i18n.localize("YZECORIOLIS.SettingRollInfoAll")
+      },
+    default: "no",
+    onChange: applyRollInfoSetting,
+  });
+
+  game.settings.register("yzecoriolis", "RollInfoArmor", {
+    name: game.i18n.localize("YZECORIOLIS.SettingRollInfoArmor"),
+    hint: game.i18n.localize("YZECORIOLIS.SettingRollInfoArmorHint"),
+    scope: "world",
+    config: true,
+    type: String,
+    choices: {
+      "no": game.i18n.localize("YZECORIOLIS.SettingRollInfoNo"),
+      "pc": game.i18n.localize("YZECORIOLIS.SettingRollInfoPC"),
+      "npc": game.i18n.localize("YZECORIOLIS.SettingRollInfoNPC"),
+      "all": game.i18n.localize("YZECORIOLIS.SettingRollInfoAll")
+      },
+    default: "no",
+    onChange: applyRollInfoSetting,
+  });
 };
+
+export function applyRollInfoSetting() {
```

Comment:
Shouldn't we register these helper in all cases?

Reply:
Actually, I think they are registered on init. I don't think we need to register them again `onChange`.

Reply:
True, works without. I will cut it here.